charts/vineyard/templates/_helpers.tpl

```diff
@@ -0,0 +1,307 @@
+      {{- end }}
+      {{- join "," $etcdEndpoint }}
+    {{- end -}}
+{{- end -}}
+
+
+{{/*
+Get the shared memory size of the worker.
+*/}}
+{{- define "vineyard.worker.size" -}}
+{{- $quota := "4Gi" -}}
+{{- range .Values.tieredstore.levels }}
+  {{- if (eq .mediumtype "MEM") }}
+    {{- $quota = .quota -}}
+    {{- break -}}
+  {{- end -}}
+{{- end -}}
+{{- $quota -}}
+{{- end -}}
+
+
+{{/*
+Check the tiered store to find whether the spill mechanism is enabled.
+*/}}
+{{- define "vineyard.checkTieredStore" -}}
+{{- $levels := index .Values "tieredstore" "levels" -}}
+{{- $memLevel := dict -}}
+{{- $ssdLevel := dict -}}
+{{- range $index, $level := $levels -}}
+  {{- if eq $level.mediumtype "MEM" -}}
+    {{- $memLevel = $level -}}
+  {{- end -}}
+  {{- if eq $level.mediumtype "SSD" -}}
+    {{- $ssdLevel = $level -}}
+  {{- end -}}
+{{- end -}}
+{{- $memLevelOk := and $memLevel (hasKey $memLevel "high") (hasKey $memLevel "low") -}}
+{{- $ssdLevelOk := and $ssdLevel (hasKey $ssdLevel "path") -}}
+{{- $valid := and $memLevelOk $ssdLevelOk -}}
+{{- $valid -}}
+{{- end -}}
+
+{{/*
+Get spill upper rate from tieredstore.
+*/}}
+{{- define "vineyard.spill.upperRate" -}}
+{{- $high := "" -}}
+{{- range .Values.tieredstore.levels -}}
+  {{- if eq .mediumtype "MEM" -}}
+    {{- $high = .high -}}
+  {{- end -}}
+{{- end -}}
+{{- $high -}}
+{{- end -}}
+
+
+{{/*
+Get spill lower rate from tieredstore.
+*/}}
+{{- define "vineyard.spill.lowerRate" -}}
+{{- $low := "" -}}
+{{- range .Values.tieredstore.levels -}}
+  {{- if eq .mediumtype "MEM" -}}
+    {{- $low = .low -}}
+  {{- end -}}
+{{- end -}}
+{{- $low -}}
+{{- end -}}
+
+
+{{/*
+Get spill path from tieredstore.
+*/}}
+{{- define "vineyard.spill.path" -}}
+{{- $path := "" -}}
+{{- range .Values.tieredstore.levels -}}
+  {{- if eq (int .level) 1 -}}
+    {{- $path = .path -}}
+  {{- end -}}
+{{- end -}}
+{{- $path -}}
+{{- end -}}
+  
+
+{{/*
+Get spill volume Mounts from tieredstore.
+*/}}
+{{- define "vineyard.worker.tieredstoreVolumeMounts" -}}
+  {{- $found := false -}}
+  {{- range .Values.tieredstore.levels }}
+    {{- if and (not $found) (eq .mediumtype "SSD") }}
+          - mountPath: {{ .path | quote }}
+            name: {{ .mediumtype | lower | quote }}
+      {{- $found = true -}}
+    {{- end }}
+  {{- end }}
+{{- end -}}
+
+
+{{/*
+Get spill volume from tieredstore.
+*/}}
+{{- define "vineyard.worker.tieredstoreVolume" -}}
+  {{- if .Values.tieredstore.levels }}
+    {{- range .Values.tieredstore.levels }}
+      {{- if eq .mediumtype "SSD" }}
+      {{- $mediumName := .mediumtype | lower }}
+      {{- if eq .type "hostPath"}}
+      - hostPath:
+          path: {{ .path }}
+          type: DirectoryOrCreate
+        name: {{ $mediumName }}
+      {{- else if eq .type "persistentVolumeClaim" }}
+      - name: {{ $mediumName }}
+        persistentVolumeClaim:
+          claimName: {{ .name }}
+      {{- else }}
+      - name: {{ $mediumName }}
+        emptyDir:
```

Comment:
I think we should consider mediumtype for emptyDir? when mediumtype=="MEM", emptyDir.meidum should be "Memory"

Reply:
Okay.